### ISSUE TITLE
Energy information

### ISSUE THREAD

**Issue by edasque** [dahlb/ha_carrier#19]
### Checklist

- [X] I have filled out the template to the best of my ability.
- [X] This only contains 1 feature request (if you have multiple feature requests, open one feature request for each feature request).
- [X] This issue is not a duplicate feature request of [previous feature requests](https://github.com/dahlb/ha_carrier/issues?q=is%3Aissue+label%3A%22Feature+Request%22+).

### Is your feature request related to a problem? Please describe.

I would love to get access to energy information from the Infinity system. I don't know if it's available through the internet "API", I see it available via Infinitude.

### Describe the solution you'd like

Adding sensors that reflect that energy information.

### Describe alternatives you've considered

Looked at Infinitude but it doesn't work with firmware 4.3x

### Additional context

Super useful integration, thank you!

**Comment by myevit** [comment]
Energy monitoring from carrier api isn't accurate. I have CT clamps in my breaker box to monitor each circus in my house including the heat pump, and I see that energy usage reported by thermostat is 1 - 1.5 kWh lower that it used in fact.
imho - no point to implement this feature.

**Comment by chapinb** [comment]
I think that reporting the data from the API can prove useful for those who don't have the CT clamps that can monitor each circuit, as without either there is no insight into usage.

I think having the prior day's usage to overlay the consumption can help identify the impact of HVAC on energy usage. Personally, it would be great to see some stats before/after adding more insulation to help quantify the impact of that project.

**Comment by adrianbn** [comment]
I think this would be useful to have even if it is not perfectly accurate. It can still be use to track trends over time.

**Comment by MallocArray** [comment]
I agree as well.  I'm coming over from Infinitude after needing to update my thermostat firmware, and glad to have found this integration.  

Even though it may not be accurate, it is more information than I have right now and since it is provided by the API, would appreciate being able to access it.

**Comment by MallocArray** [comment]
I wish I could help develop this, but not sure that I would be able to.  Would a donation help give this some attention, even if it isn't perfectly accurate?

**Comment by dahlb** [comment]
donations wouldn't affect my attention but maybe you could attract another contributor.

I've never seen a carrier api that provides the energy use, if you find another integration that uses carrier's api to provide it we could port it.

It doesn't seem to be available through the web interface aka https://myinfinitytouch.carrier.com/account

**Comment by MallocArray** [comment]
Ah, that is a different thing in that case.

I'm not finding it in the web interface either, nor do I see any network traffic from the site that looks to include it.

I am able to see it in the Carrier Home Android App by clicking on the Gear next to my location name and selecting "Energy Usage">Manage, but if that isn't available in the standard page results, then maybe it can't be retrieved.

When I was using Infinitude, there was a /energy option to retrieve it, so was hoping to still have access.

**Comment by dahlb** [comment]
@hypomodern which web interface are you seeing this in? I normally use https://www.myinfinitytouch.carrier.com/account but don't see this there

**Comment by hypomodern** [comment]
Sure yep! That's an alias for https://my.carrier.com . When I log in there, I can select my device in the top row, which opens a side panel, click on the gear icon in the top right of that panel, then on "Status and Usage"

<img width="507" alt="Image" src="https://github.com/user-attachments/assets/e6a41ead-9f50-4d2b-b223-aa088af88231" />

**Comment by dahlb** [comment]
I'm going to open a new ticket to track attempting to switch to the new graphql api, but investigating the new vs old api I found the energy end point in the old api the integration is currently using and has exposed yearly usage sensors in v1.29.0

**Comment by MallocArray** [comment]
Is yearly usage the only option, or is there the ability to see the last day or 2 days, similar to what it seen on the physical thermostat?

**Comment by dahlb** [comment]
If you configure the home assistant energy dashboard to use the yearly it will let you see if hourly with history and graphing
